istioctl/pkg/util/configdump/wrapper.go

```diff
@@ -19,8 +19,21 @@ import (
 
 	adminapi "github.com/envoyproxy/go-control-plane/envoy/admin/v2alpha"
 	"github.com/gogo/protobuf/jsonpb"
+	proto "github.com/gogo/protobuf/proto"
 )
 
+// Allow Istio 1.3 Envoy with `"@type": "type.googleapis.com/envoy.admin.v2alpha.ScopedRoutesConfigDump"`
+// to work with 1.2 "istioctl proxy-config".
+type dummy struct{}
+
+func init() {
+	proto.RegisterType((*dummy)(nil), "envoy.admin.v2alpha.ScopedRoutesConfigDump")
```

Comment:
jsonpb allows setting `AllowUnknownFields` I believe. I think this would be a more robust solution (if istioctl is using that?). A similar PR was done here https://github.com/istio/istio/pull/13699/files#diff-a81ffcf40edc6cb8320783f666dc6fe9.

This way we would be protected from future breakages as well.

Reply:
I agree; I also want _istioctl_ to ignore unknown types.

We already ignore unknown fields using `AllowUnknownFields: true`.  https://github.com/istio/istio/blob/2d359e5b1fc934a66558cee0c71c547ac0f928a8/istioctl/pkg/util/configdump/wrapper.go#L43

This is a different problem.  It's an unknown type.  https://github.com/golang/protobuf/issues/747 says a custom AnyResolver is needed.  I will look into that approach.

Reply:
OK, I bit the bullet and now ignore all unknown types.  Please review the line `return &exprpb.Type{TypeKind: &exprpb.Type_Dyn{Dyn: &emptypb.Empty{}}}, nil` which works although I am not clear why I had to do that unstead of just returning emptypb.Empty as suggested by the protobuf issue 747.